amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutRecordsPublisher.java

```diff
@@ -158,20 +151,26 @@ RecordFlow evictAckedEventAndScheduleNextEvent(RecordsDeliveryAck recordsDeliver
         // RecordFlow of the current event that needs to be returned
         RecordFlow flowToBeReturned = null;
 
+        final RecordsRetrieved recordsRetrieved = recordsRetrievedContext != null ?
+                recordsRetrievedContext.getRecordsOrShutDownEvent()
+                        .map(recordsEvent -> recordsEvent, shutDownEvent -> null) : null;
```

Comment:
let's move this logic out into RecordsRetrievedContext. getRecordsRetrieved either returns null or records depending on the event type stored in the context.